lib/cocoapods-core/source.rb

```diff
@@ -266,6 +267,11 @@ def search(query)
         query = query.root_name
       end
 
+      if (versions = @versions_by_name[query]) && !versions.empty?
```

Comment:
if we already have a cached list of versions for this pod, skip globbing the pod directory to retrieve the list of versions